src/serde/tree_cache.rs

```diff
@@ -95,7 +93,11 @@ pub struct TreeCache {
     /// node_entries vector. For any given tree hash, we're only supposed to
     /// have a single NodeEntry. There may be multiple NodePtr referring to
     /// the same NodeEntry (if they are identical sub trees).
-    hash_to_node: HashMap<Bytes20, u32, RandomState>,
+    atom_lookup: HashMap<Bytes20, u32, RandomState>,
+
+    /// maps left + right child indices to the index of the pair with those
+    /// children. This is the atom_lookup counterpart for pairs
+    pair_lookup: HashMap<u64, u32>,
```

Comment:
Any reason not write it as the (IMO more clear) `(u32, u32)` ?

Who knows how rust would encode it. Are you concerned it would be slower?